src/PerformanceTests/Common/Scenarios/BaseRunner.cs

```diff
@@ -214,4 +224,32 @@ protected virtual Task Setup()
         await Task.Delay(Settings.RunDuration).ConfigureAwait(false);
         Log.Info("Run duration expired.");
     }
-}
\ No newline at end of file
+
+    static Random random = new Random();
+    async Task RetryWithBackoff(Func<Task> action, CancellationToken token, string id, int maxAttempts)
+    {
+        while (true)
+        {
+            var attempts = 0;
+            try
+            {
+                await action()
+                    .ConfigureAwait(false);
+                return;
+            }
+            catch (Exception ex)
+            {
+                if (attempts > maxAttempts) throw new InvalidOperationException("Exhausted send retries.", ex);
+                double next;
+                lock (random) next = random.NextDouble();
+                next *= 0.2; // max 20% jitter
```

Comment:
I don't understand why you want to add that. What's the value of this "jitter"?

Reply:
The jitter is to randomize the delays. If you get throttle, imagine 5,000 items get delayed at the same time. If you sleep them all for the same duration you will likely get throttled immediately again. This spreads the retries. Especially over multiple retries.